File: cpp/Trawa.cpp
#include "Trawa.h"
#include "Komentarz.h"

Trawa::Trawa(Swiat *swiat, Punkt pole, int turaPowstania)
    : Roslina(swiat, pole, turaPowstania, sila, inicjatywa, symbol){
    this->symbol = SYMBOL_TRAWY;
    this->sila = SILA_TRAWY;
    this->inicjatywa = INICJATYWA_TRAWY;
    this->pole = pole;
    this->swiat = swiat;
    this->umarl = false;
    this->turaPowstania = turaPowstania;
}

string Trawa::nazwaOrganizmu() {
    return "Trawa";
}

File: java/pl/edu/pg/po/projekt2/rosliny/Trawa.java
package pl.edu.pg.po.projekt2.rosliny;

import pl.edu.pg.po.projekt2.Punkt;
import pl.edu.pg.po.projekt2.Roslina;
import pl.edu.pg.po.projekt2.Swiat;

public class Trawa extends Roslina {
    private static final char symbol = 'T';
    private static final int inicjatywa = 0;
    private static final int sila = 0;

    public Trawa(Swiat swiat, Punkt pole, int turaPowstania) {
        super(swiat, pole, turaPowstania, sila, inicjatywa, symbol);
        this.pole = pole;
        this.swiat = swiat;
        this.umarl = false;
        this.turaPowstania = turaPowstania;
    }

    @Override
    public String nazwaOrganizmu() {
        return "Trawa";
    }
}
